src/com/palmergames/bukkit/towny/command/PlotCommand.java

```diff
@@ -1607,33 +1606,39 @@ private boolean handlePlotGroupCommand(String[] split, Player player) throws Tow
-				
 				Runnable permHandler = () -> {
-
-					// setTownBlockPermissions returns a towny permission change object
-					TownyPermissionChange permChange = PlotCommand.setTownBlockPermissions(player, townBlockOwner, townBlock, StringMgmt.remArgs(split, 2));
-					// If the perm change object is not null
-					if (permChange != null) {
-						plotGroup.getPermissions().change(permChange);
-						
-						plotGroup.getTownBlocks().stream()
-							.forEach(tb -> {
-								tb.setPermissions(plotGroup.getPermissions().toString());
-								tb.setChanged(!tb.getPermissions().toString().equals(town.getPermissions().toString()));
-								tb.save();
-								// Change settings event
-								BukkitTools.fireEvent(new TownBlockSettingsChangedEvent(tb));
-							});
-
-						plugin.resetCache();
+					try {
+						// setTownBlockPermissions returns a towny permission change object
+						TownyPermissionChange permChange = PlotCommand.setTownBlockPermissions(player, townBlockOwner, townBlock, StringMgmt.remArgs(split, 2));
+						// If the perm change object is not null
+						if (permChange != null) {
+							plotGroup.getPermissions().change(permChange);
+							plotGroup.getTownBlocks().stream()
+								.forEach(tb -> {
+									try {
+										String permissions = plotGroup.getPermissions().toString();
+										BukkitTools.ifCancelledThenThrow(new TownBlockPermissionChangeEvent(townBlock, permissions));
+										tb.setPermissions(permissions);
+										tb.setChanged(!tb.getPermissions().toString().equals(town.getPermissions().toString()));
+											
+										tb.save();
+										// Change settings event
+										BukkitTools.fireEvent(new TownBlockSettingsChangedEvent(tb));
+									} catch (CancelledEventException ignore) { }
+								});
+	
+							plugin.resetCache();
+							
+							TownyPermission perm = plotGroup.getPermissions();
+							TownyMessaging.sendMessage(player, Translatable.of("msg_set_perms").forLocale(player));
+							TownyMessaging.sendMessage(player, (Colors.Green + Translatable.of("status_perm").forLocale(player) + " " + ((townBlockOwner instanceof Resident) ? perm.getColourString().replace("n", "t") : perm.getColourString().replace("f", "r"))));
+							TownyMessaging.sendMessage(player, Colors.Green + Translatable.of("status_pvp").forLocale(player) + " " + (!(CombatUtil.preventPvP(townBlock.getWorld(), townBlock)) ? Colors.Red + "ON" : Colors.LightGreen + "OFF") + 
+																Colors.Green + Translatable.of("explosions").forLocale(player) + " " + ((perm.explosion) ? Colors.Red + "ON" : Colors.LightGreen + "OFF") + 
+																Colors.Green + Translatable.of("firespread").forLocale(player) + " " + ((perm.fire) ? Colors.Red + "ON" : Colors.LightGreen + "OFF") + 
+																Colors.Green + Translatable.of("mobspawns").forLocale(player) + " " + ((perm.mobs) ? Colors.Red + "ON" : Colors.LightGreen + "OFF"));
+						}
+					} catch (CancelledEventException e) {

```

Comment:
If we're going to actually keep the try/catch here we need to use it to display the cancellation message.